pkg/sdn/plugin/registry.go

```diff
@@ -263,6 +263,34 @@ func (registry *Registry) GetEgressNetworkPolicies() ([]osapi.EgressNetworkPolic
 	return policyList.Items, nil
 }
 
+// Returns true to indicate status update is needed
+type UpdateNodeFunc func(node *kapi.Node) bool
+
+func (registry *Registry) UpdateNodeStatus(node *kapi.Node, updateFunc UpdateNodeFunc) (*kapi.Node, error) {
```

Comment:
in either case, making this just update the node status and moving the RetryOnConflict to the caller, rather than passing in an UpdateNodeFunc, would simplify this